cmd/cluster/resize/infra_node.go

```diff
@@ -155,6 +228,35 @@ func (r *Resize) RunInfra(ctx context.Context) error {
 		return err
 	}
 
+	// Wait for original nodes to delete
+	if err := wait.PollImmediate(twentySecondIncrement, twentyMinuteTimeout, func() (bool, error) {
+		// Re-check for originalNodes to see if they have been deleted
+		return r.nodesMatchExpectedCount(ctx, originalNodeSelector, 0)
+	}); err != nil {
+		switch {
+		case err == wait.ErrWaitTimeout:
```

Comment:
Fun fact about Go - comparing errors for equality with `==` will fail on wrapped errors. It's usually preferred to use `errors.Is`
```suggestion
		errors.Is(err, wait.ErrWaitTimeout):
```

Reply:
TIL, thanks, fixed in next push.